src/libtorch.cc

```diff
@@ -1284,62 +1569,113 @@ ModelInstanceState::ReadOutputTensors(
-    for (auto itr = shape.begin(); itr != shape.end(); itr++) {
-      batchn_shape.push_back(*itr);
-    }
+      // Get output shape
+      std::vector<int64_t> batchn_shape;
+      auto shape = output_tensors[op_index].toTensor().sizes();
+      for (auto itr = shape.begin(); itr != shape.end(); itr++) {
+        batchn_shape.push_back(*itr);
+      }
+
+      if (batchn_shape.size() == 0) {
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            (std::string("output '") + name +
+             "' is a scalar which is not supported.")
+                .c_str());
+      }
+
+      responder.ProcessTensor(
+          name, output_dtype, batchn_shape, output_buffer,
+          (tensor_device.type() == torch::kCPU) ? TRITONSERVER_MEMORY_CPU
+                                                : TRITONSERVER_MEMORY_GPU,
+          (tensor_device.type() == torch::kCPU) ? 0 : tensor_device.index());
+
+    } else if (output_tensors[op_index].isList()) {
+        // Custom handling for string/bytes tensor...
+
+      torch::List<torch::jit::IValue> output_list = output_tensors[op_index].toList();
+
+      // Get output shape
+      size_t list_length = output_list.size();
+      std::vector<int64_t> batchn_shape{list_length};
+      if (output_list.elementType()->kind() != c10::TypeKind::StringType) {
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            (std::string("output '") + name +
+             "' must be of type Tensor or List[str], recieved List[" +
+             output_list.elementType()->str() + "]")
+                .c_str());
+      }
+
+      size_t tensor_offset = 0;
 
-    if (batchn_shape.size() == 0) {
-      RETURN_IF_ERROR(TRITONSERVER_ErrorNew(
+      for (size_t idx = 0; idx < responses->size(); idx++) {
+        auto& response = (*responses)[idx];
+
+        const size_t tensor_element_cnt = GetElementCount(batchn_shape);
+
+        // Only need an response tensor for requested outputs.
+        if (response != nullptr) {
+          TRITONBACKEND_Output* response_output;
+          RESPOND_AND_SET_NULL_IF_ERROR(
+              &response, TRITONBACKEND_ResponseOutput(
+                             response, &response_output, name.c_str(), TRITONSERVER_TYPE_BYTES,
+                             batchn_shape.data(), batchn_shape.size()));
+          string_buffer.emplace_back(new std::string());
+          cuda_copy |= SetStringOutputBuffer(
+              &output_list, &response, response_output, tensor_element_cnt,
+              tensor_offset, CudaStream(), string_buffer.back().get());
+        }
+
+        tensor_offset += tensor_element_cnt;
+      }
+    } else {
+      return TRITONSERVER_ErrorNew(
           TRITONSERVER_ERROR_INVALID_ARG,
           (std::string("output '") + name +
-           "' is a scalar which is not supported.")
-              .c_str()));
+           "' must be of type Tensor or List[str].")
```

Comment:
We must throw an error when loading the model.
1. If the dynamic_batching is enabled for string input.
2. If more than one dimension is specified for TYPE_STRING.